### ISSUE TITLE
Unable To Rename Items In Library View

### ISSUE THREAD

**Issue by ghost** [derceg/explorerplusplus#140]
Windows 10 1803.17134.590, Explorer++ 1.4.0.620 (behaviour exhibited on 1.3.5 as well)
Occurs both when running as Administrator and also as a regular user.

When using a library and attempting to rename files or folders, I receive an error stating it cannot be renamed, possibly due to being in use. This occurs despite the fact the file is able to be edited outside of library view just fine, [video here](https://i.imgur.com/JvBM5tY.gifv).

When attempting to copy a folder within a library to another part of the library, I receive [this error message](https://i.imgur.com/PyLBKlx.png) stating that the folder does not exist. The name it gives is not the name of the folder in question, but (at a guess) some internal name. This behaviour does not occur if outside of library view and attempting to move the same folder.

A few users in [this thread](https://explorerplusplus.com/forum/viewtopic.php?f=5&t=1029&start=10#p3764) made reference to the issue, but it appears to have been overlooked since it wasn't in the scope of the thread topic.

---

Unfortunately my system claims explorer++ does not exist, and therefore prevents me from enabling logging, even if I manually add it to my PATH. It's most likely user error, but either way, I'm unable to see if the logs would help - although on that note, giving an option in Tools to enable logging would be beneficial to users for bug reports, I'm sure. Any assistance on figuring that out would be appreciated to better help with this bug report, but isn't necessary due to how seemingly easy it is to replicate.

**Comment by derceg** [comment]
Renaming items in a library folder should now work. You can now delete items and create new folders as well.

Note that because Explorer++ doesn't currently monitor virtual directories for changes, you'll have to manually refresh the folder after adding/renaming/deleting a file. I started doing some work last year to fix that, but it's still in progress and none of the changes are in the main development version.

**Comment by Nomuit** [comment]
~~Unfortunately the latest development build doesn't appear to have fixed the issue, if the commit above was included in it (1.4.0.666), [video here](https://i.imgur.com/HwVWSCV.mp4). In the event it matters, I have my Explorer++ folder within Program Files, however the issue also occurs when using it directly from Downloads.~~
~~So it appears that the location of Explorer++ plays a role in this. I tend to keep programs in Program Files or Program Files (x86), and it didn't occur to me that may be causing the strange behaviour, as I'd tested both running as Administrator and as a User. Just in case location* was affecting it I tried it in a different directory as well, but I forgot Downloads tends to be sandboxed. Upon trying it elsewhere, I discovered the behaviour doesn't occur.~~
Nope, I'm wrong again. Some virtual folders are able to be renamed regardless of application directory, and others aren't able to be renamed regardless of application directory (or File Explorer replacement level, or Administrator/User). [Video here](https://i.imgur.com/wq27NtW.mp4). Assuming the commit above was included in 1.4.0.666, then there's something else at play. I'm not sure if it's file/folder permissions or if it's due to some folders being OS generated or what, but I can't tell on my end.

As for manually refreshing, I don't particularly mind that honestly. It'd be a nice improvement of course, but I'm perfectly okay with hitting F5 after edits, and it's not nearly as disruptive as being unable to work in libraries at all *(if that comment was due to the thread I linked, I was only in the thread due to it turning up as a result upon searching for this issue)*.
